test/spotify_sdk_test.dart

```diff
@@ -1,26 +1,69 @@
+import 'dart:convert';
+
 import 'package:flutter/services.dart';
 import 'package:flutter_test/flutter_test.dart';
+import 'package:spotify_sdk/models/crossfade_state.dart';
+import 'package:spotify_sdk/platform_channels.dart';
 import 'package:spotify_sdk/spotify_sdk.dart';
 
 void main() {
-  const channel = MethodChannel('spotify_sdk');
+  const channel = MethodChannel(MethodChannels.spotifySdk);
 
   TestWidgetsFlutterBinding.ensureInitialized();
 
-  setUp(() {
-    channel.setMockMethodCallHandler((MethodCall methodCall) async {
-      return true;
-    });
-  });
+  setUp(() {});
 
   tearDown(() {
     channel.setMockMethodCallHandler(null);
   });
 
-  test('connectToSpotify', () async {
-    expect(
-        await SpotifySdk.connectToSpotifyRemote(
-            clientId: 'null', redirectUrl: 'null'),
-        true);
+  void setupMethodCall(String methodCall) {
+    channel.setMockMethodCallHandler((call) async {
+      switch (call.method) {
+        case MethodNames.connectToSpotify:
+          return true;
+        case MethodNames.getAuthenticationToken:
+          return MethodNames.getAuthenticationToken;
+        case MethodNames.getCrossfadeState:
+          var crossfadeState = CrossfadeState(12, isEnabled: true);
+          var json = jsonEncode(crossfadeState);
+          return json;
+      }
+    });
+  }
+
+  group(MethodNames.connectToSpotify, () {
+    test('withValidArguments', () async {
+      setupMethodCall(MethodNames.connectToSpotify);
+
+      expect(
+          await SpotifySdk.connectToSpotifyRemote(
+              clientId: "089d841ccc194c10a77afad9e1c11d54",
+              redirectUrl: "comspotifytestsdk://callback"),
+          true);
+    });
+  });
+
+  group(MethodNames.getAuthenticationToken, () {
+    test('withValidArguments', () async {
+      setupMethodCall(MethodNames.getAuthenticationToken);
+
+      expect(
+          await SpotifySdk.getAuthenticationToken(
+              clientId: "089d841ccc194c10a77afad9e1c11d54",
+              redirectUrl: "comspotifytestsdk://callback",
+              scope: ""),
```

Comment:
```suggestion
          await SpotifySdk.getAuthenticationToken(clientId: null, redirectUrl: null, scope: null),
```

The arguments do not seem offer anything. If I set them to null the test still succeeds. Can we actually test the arguments somehow?

Reply:
Good question, I am testing this right now, But wanted to make a foundation for our tests. In general this PR is WIP right now